src/Servers/Kestrel/Transport.NamedPipes/src/Internal/NamedPipeConnectionListener.cs

```diff
@@ -26,7 +26,7 @@ internal sealed class NamedPipeConnectionListener : IConnectionListener
     private readonly PipeOptions _inputOptions;
     private readonly PipeOptions _outputOptions;
     private readonly Mutex _mutex;
-    private Task? _listeningTask;
+    private Task[]? _listeningTasks;
```

Comment:
Could be simplified, using `Task.WhenAll`, as it is awaited on a non-critical path

Reply:
Actually Task.WhenAll is used on DisposeAsync. Not sure what is the best option here.

Reply:
I don't think it really matters. A connection listener is created once when the server starts. Not concerned about what it allocates.

Reply:
I'm surprised it doesn't matter.

Reply:
These tasks are awaited when the connection listener shutdowns. Whether `Task.WhenAll` aggregates the tasks on startup or shutdown doesn't matter*

* Ok, there is one extra `Task` allocation for the app's lifetime for calling `Task.WhenAll` on startup. That's why the PR currently leaves it to shutdown.